lib/ecto/adapters/mssql/connection.ex

```diff
@@ -0,0 +1,519 @@
+      end)
+    end
+
+    def execute_ddl({:create, %Index{}=index}) do
+      IO.inspect index.columns
+      assemble(["CREATE#{if index.unique, do: " UNIQUE"} INDEX",
+                index.name, " ON ", index.table,
+                " (#{Enum.map_join(index.columns, ", ", &index_expr/1)})"])
+    end
+
+    def execute_ddl({:drop, %Index{}=index}) do
+      assemble(["DROP INDEX", index.name, " ON ", index.table])
+    end
+
+    def execute_ddl(default) when is_binary(default), do: default
+
+    defp column_definitions(columns) do
+      Enum.map_join(columns, ", ", &column_definition/1)
+    end
+
+    defp column_definition({:add, name, type, opts}) do
+      assemble([name, column_type(type, opts), column_options(opts), serial_expr(type)])
+    end
+
+    # defp column_changes(columns) do
+    #   Enum.map_join(columns, ", ", &column_change/1)
+    # end
+
+    defp column_change({:add, name, type, opts}) do
+      assemble(["ADD", name, column_type(type, opts), column_options(opts)])
+    end
+
+    defp column_change({:modify, name, type, opts}) do
+      assemble(["ALTER COLUMN", name, column_type(type, opts)])
+    end
+
+    defp column_change({:remove, name}), do: "DROP COLUMN #{name}"
+
+    defp column_options(opts) do
+      default = Keyword.get(opts, :default)
+      null    = Keyword.get(opts, :null)
+      pk      = Keyword.get(opts, :primary_key)
+
+      [default_expr(default), null_expr(null), pk_expr(pk)]
+    end
+
+    defp pk_expr(true), do: "PRIMARY KEY"
+    defp pk_expr(_), do: nil
+
+    defp serial_expr(:serial), do: "IDENTITY"
+    defp serial_expr(_), do: nil
+
+    defp null_expr(false), do: "NOT NULL"
+    defp null_expr(true), do: "NULL"
+    defp null_expr(_), do: nil
+
+    defp default_expr(nil),
+      do: nil
+    defp default_expr(literal) when is_binary(literal),
+      do: "DEFAULT '#{escape_string(literal)}'"
+    defp default_expr(literal) when is_number(literal),
+      do: "DEFAULT #{literal}"
+    defp default_expr({:fragment, expr}),
+      do: "DEFAULT #{expr}"
+
+    defp index_expr(literal) when is_binary(literal),
+      do: literal
+    defp index_expr(literal),
+      do: literal
+
+    defp column_type(%Reference{} = ref, opts) do
+      "bigint REFERENCES #{ref.table}(#{ref.column})"
+    end
+    defp column_type({:array, type}, opts),
+      do: "nvarchar(max)"
+    defp column_type(:uuid, opts), do: "uniqueidentifier"
+    defp column_type(type, opts) do
+      pk        = Keyword.get(opts, :primary_key)
+      size      = Keyword.get(opts, :size)
+      precision = Keyword.get(opts, :precision)
+      scale     = Keyword.get(opts, :scale)
+      type_name = ecto_to_db(type)
+
+      cond do
+        type == :serial -> "bigint"
+        pk == true      -> "bigint"
+        size            -> "#{type_name}(#{size})"
+        precision       -> "#{type_name}(#{precision},#{scale || 0})"
+        type == :string -> "#{type_name}(255)"
+        type == :binary -> "#{type_name}(255)"
```

Comment:
This feels wrong. Is there a blob or something similar in SQLServer that does not contain a size?